pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -222,15 +222,17 @@ func (pl *DefaultPreemption) SelectVictimsOnNode(
 	return victims, numViolatingVictim, framework.NewStatus(framework.Success)
 }
 
-// PodEligibleToPreemptOthers determines whether this pod should be considered
-// for preempting other pods or not. If this pod has already preempted other
+// PodEligibleToPreemptOthers returns one bool and one string. The bool
+// indicates whether this pod should be considered for preempting other pods or
+// not. The string includes the reason if this pod isn't eligible.
+// If this pod has a preemptionPolicy of Never or has already preempted other
 // pods and those are in their graceful termination period, it shouldn't be
 // considered for preemption.
 // We look at the node that is nominated for this pod and as long as there are
 // terminating pods on the node, we don't consider this for preempting more pods.
 func (pl *DefaultPreemption) PodEligibleToPreemptOthers(pod *v1.Pod, nominatedNodeStatus *framework.Status) (bool, string) {
 	if pod.Spec.PreemptionPolicy != nil && *pod.Spec.PreemptionPolicy == v1.PreemptNever {
-		return false, fmt.Sprintf("pod is not eligible for preemption because it has a preemptionPolicy of Never.")
+		return false, fmt.Sprint("not eligible for preemption due to preemptionPolicy=Never.")
```

Comment:
if we add the word preemption somewhere else, perhaps we can remove the word from here.

Reply:
If we use <postFilter status> to replace <preemption status> elsewhere, the preemption is needed here.

Reply:
I prefer that we add the string "preemption:" somewhere else, like right before returning the PostFilter result https://github.com/kubernetes/kubernetes/blob/f49a15ae50196fc4d46d70f86de4d15cbc2405c6/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go#L98
Or perhaps inside the Preempt function?

Reply:
like this？ feels a little strange.
```go
	result, status := pe.Preempt(ctx, pod, m)
	return result, framework.NewStatus(status.Code(), "preemption:"+status.Message())
```

Reply:
yeah, something like that... probably needs a space

Reply:
oh, and the reason would be something like: "not eligible due to X"

Reply:
> oh, and the reason would be something like: "not eligible due to X"

not only `not eligible due to X`, also could be like this `0/4 nodes are available: 1 Insufficient cpu, 3 Preemption is not helpful for scheduling.`

> yeah, something like that... probably needs a space
addressed.

Reply:
> Can you remove the word "preemption" in the messages inside PodEligibleToPreemptOthers?

Done

> Also please update how the messages look like in the PR description.

Sorry, I forgot it. Updated.